### DISCUSSION TITLE
(optional) automatic query batching and cache?

### DISCUSSION THREAD

**Discussion by FreekVR** [Diizzayy/nuxt-graphql-client#403] [Ideas]
Hi there!

This library looks great, and I am looking into the viability of this replacing our our GraphQL nuxt client which is a wrapper around [villus](https://villus.logaretm.com/)

However, two features I am fond of with Villus are the automagic query batching and cache support.

I know manual query batching is possible with graphql-request, but would this module support automatic batching as villus does? If I understand their implementation correctly, you can set an optional (minimal) timeout and it will automatically gather what would otherwise be multiple requests, put them together in a batch, and then send the individual results back to the callee as if it was just one request originally.

Similarly, it does some smart caching by storing any existing queries in memory, and validating any future queries against this cache by comparing the values of the variables and the query.

Loving the codegen and DX of this module compared to villus, but perf-wise would be great to have these features as an option here, too.

**Comment by davidwebca** [comment]
I've upvoted this a long time ago, but I just wanted to manifest my interest in this too again as I'm hitting performance issues due to sending multiple requests at the same time. @FreekVR in the end, have you fully moved to nuxt-graphql-client or have you kept your villus wrapper in the end?
